### ISSUE TITLE
About the purpose of Webflux

### ISSUE THREAD

**Issue by HelloTo4x** [AlteracFinished/webflux-backend#1]
Hi, i am reading your project, can u tell me the purpose of Webflux, reactive in your project ? Is it for increasing performance, can u tell me some disadvantaged of Webflux ?

**Comment by AlteracFinished** [comment]
This project is just a sample of reactive applications implemented by Spring. 
Webflux and [Reactive Relational Database Connectivity](https://r2dbc.io/) can avoid JDBC driver blocking, improving the application's throughput. ([Reference blog here](https://technology.amis.nl/software-development/performance-and-tuning/spring-blocking-vs-non-blocking-r2dbc-vs-jdbc-and-webflux-vs-web-mvc/)). Using the Webflux only cannot bring significant performance improvement.
The disadvantages of Webflux are as follows:
1. The developers have to know precisely how to deal with reactor objects(Mono and Flux). It could be problematic for developers who are not familiar with it.
2. The application would have to find its reactive connection library with other data sources like Redis and Consul. It is not easy to find compatible reactive libraries for each data source.

In short, if high concurrency is not required, it could be more convenient and stable to use Web MVC and JDBC.

**Comment by HelloTo4x** [comment]
thank @AlteracFinished , but why do you use Redis, why don't you use other cache like Ehcache, and about Consul, you use it for discovering service ?, why don't you use Eureka or Zookeeper

**Comment by AlteracFinished** [comment]
> thank @AlteracFinished , but why do you use Redis, why don't you use other cache like Ehcache, and about Consul, you use it for discovering service ?, why don't you use Eureka or Zookeeper

Ehcache is a good choice for the application because using Redis for this sample project would be a little complicated. But with Redis, we can easily share the user session with other applications. And if I want to use Ehcache for user session management, I have to implement it myself because Ehcache is not supported by the spring-data-session project.

I have used Zookeeper, Consul and Nacos. [Because the spring-cloud-netflix project is under maintenance mode,](https://cloud.spring.io/spring-cloud-netflix/multi/multi__modules_in_maintenance_mode.html) I am not trying to use Eureka by now. For me, Zookeeper and Consul are good choices for service discovery, but Nacos also provides externalized configuration and service flow control, which is very helpful when I want to reduce the system dependencies.

**Comment by HelloTo4x** [comment]
thank u so much
